### PR TITLE
Add sphinx doc

### PR DESCRIPTION
**Pull request by JoseAlanis**
Hey guys,

I'm adding the sphix configuration files for building the documentation.
Should I build the doc locally first and an then push the changes, i.e., add `_build` dir, for intance? Or will these be created later?

The rendered index site looks like this (descriptions follow):
mne-stats

### CODE REVIEW

**Comment by JoseAlanis** [comment]
Updated doc page:

Bildschirmfoto 2019-07-11 um 12 59 28

**Comment by dengemann** [comment]
Looks already pretty awesome!

**Comment by dengemann** [comment]
Looks already pretty awesome! Taking detailed look later.

File: doc/conf.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
+# -*- coding: utf-8 -*-
+#
+# autoreject documentation build configuration file, created by
+# sphinx-quickstart on Mon May 23 16:22:52 2016.
```
Comment (dengemann) [inline doc/conf.py]: watchout you've got a few of these. Make sure you don't refer to other projects here.

File: doc/conf.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+# If false, no module index is generated.
+# latex_domain_indices = True
+
+
+# -- Options for manual page output ---------------------------------------
+
+# One entry per manual page. List of tuples
+# (source start file, name, description, authors, manual section).
+man_pages = [
+    (master_doc, 'MNE-Stats', u'MNE-Stats Documentation',
```
Comment (dengemann) [inline doc/conf.py]: stats->limo ?

File: doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+In addition, we have started to develop method to translate single subject analysis to group-levels analysis, i.e., estimating linear regression effects over a series of subjects.
+Please visit the :ref:`group level analysis gallery <sphx_glr_limo_examples_group_level>` for more information on how to carry out linear regression analysis on a group level.
+
+
+Acknowledgements
+================
+
+This project is currently supported by a `2019 Google Summer of Code project <https://summerofcode.withgoogle.com/projects/#5715889406607360>`_ grant issued to José C. García Alanis.
+
+Special acknowledgements go to Denis A. Engemann, Jona Sassenhagen, and the MNE-Community for their support, guidance, and inputs through the project.
```
Comment (dengemann) [inline doc/index.rst]: "throughout" or "for" the project.

**Comment by dengemann** [comment]
other than my few comments looks good! Is it building properly and is it writing into `docs`, such that pushing to master / updating the website will be easy?

File: doc/conf.py (hunk 1 above)
Comment (JoseAlanis) [reply doc/conf.py]: oups, thanks for the hint (do you see any other false references, I think I removed everything else). Also, the file I used as reference template was from the mne-hcp repo. So perhaps we should have a look at that repo as well.

**Comment by JoseAlanis** [comment]
> other than my few comments looks good! Is it building properly and is it writing into `docs`, such that pushing to master / updating the website will be easy?

Yes, it builds without problems on my machine and wrtites the files into the `/doc` dir.

**Comment by dengemann** [comment]
please merge once you are happy

**Pull request merged**